mods/cnc/maps/nod07c/nod07c.lua

```diff
@@ -0,0 +1,199 @@
+		Utils.Do(actors, function(actor)
+			if actor.IsIdle then
+				actor.Patrol(team.waypoints, true, DateTime.Seconds(25))
+			end
+		end)
+	end
+end
+
+Trigger.OnKilled(Gunboat, function()
+	GunboatCamera.Destroy()
+end)
+
+Trigger.OnKilled(GDIHpad, function()
+	if not player.IsObjectiveCompleted(NodObjective1) then
+		player.MarkFailedObjective(NodObjective1)
+	end
+end)
+
+Trigger.OnKilled(GDIOrca, function()
+	if not player.IsObjectiveCompleted(NodObjective2) then
+		player.MarkFailedObjective(NodObjective2)
+	end
+end)
+
+Trigger.OnDamaged(GDIBuilding11, function()
+	SendGuards(MTANK)
+end)
+ 
+Utils.Do(Map.ActorsWithTag("Village"), function(actor)
+	Trigger.OnKilled(actor, function()
+		targetsKilled = targetsKilled + 1
+	end)
+end)
+
+Trigger.OnCapture(GDIHpad, function()
+	hpadCaptured = true
+	player.MarkCompletedObjective(NodObjective1)
+	if not GDIOrca.IsDead then
+		GDIOrca.Owner = player
+	end
+	Actor.Create("camera", true, { Owner = player, Location = waypoint25.Location })
+	Actor.Create("flare", true, { Owner = player, Location = waypoint25.Location })
+end)
+
+WorldLoaded = function()
+	player = Player.GetPlayer("Nod")
+	enemy = Player.GetPlayer("GDI")
+
+	Camera.Position = waypoint26.CenterPosition
+
+	GunboatCamera = Actor.Create("camera.boat", true, { Owner = player, Location = Gunboat.Location })
+	InsertNodUnits()
+
+	SendGuards(GDI1)
+	SendGuards(GDI3)
+	AutoGuard(enemy.GetGroundAttackers())
+
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerWon(player, function()
+		Media.PlaySpeechNotification(player, "Win")
+	end)
+
+	Trigger.OnPlayerLost(player, function()
+		Media.PlaySpeechNotification(player, "Lose")
+	end)
+
+	Trigger.OnIdle(Gunboat, function() Gunboat.Patrol(GunboatPatrolPath) end)
+
+	Trigger.OnDiscovered(GDIBuilding9, DiscoveredMainEntrance)
+	Trigger.OnDiscovered(GDIBuilding10, DiscoveredMainEntrance)
+	Trigger.OnDiscovered(GDIBuilding11, DiscoveredSideEntrance)
+
+	NodObjective1 = player.AddPrimaryObjective("Capture the Helipad.")
+	NodObjective2 = player.AddPrimaryObjective("Destroy the village with the GDI orca.")
+	NodObjective3 = player.AddSecondaryObjective("Use your vehicles to attack the front entrance\nand distract the guards.")
+	GDIObjective = enemy.AddPrimaryObjective("Kill all enemies.")
+end
+
+Tick = function()
+	if not Gunboat.IsDead then
+		GunboatCamera.Teleport(Gunboat.Location)
+	end
+
+	if DateTime.GameTime > 2 and player.HasNoRequiredUnits() then
+		enemy.MarkCompletedObjective(GDIObjective)
+	end
+
+	if targetsKilled >= 10 then
```

Comment:
I think this is a bit weird, and leads to situations like: (mission is "Accomplished", but most of the town still stands)
![openra-2016-06-05t110640z](https://cloud.githubusercontent.com/assets/7704140/15806284/6873a332-2b40-11e6-8279-4facc45cf84d.png)
(although building transports is now disabled, you could still achieve that with only the orca).